Program.cs

```diff
@@ -101,29 +102,59 @@ private static void WriteToDss(string interval, DateTime t, DssWriter dss, int s
         parameter = "Temperature";
       }
       string dssPath = "/Trinity/" + dsn + "/" + parameter + "//" + interval + "/" + F + "/";
-      WriteToDss(dss, data, dssPath, t, units, dataType);     
+      WriteToDss(dss, data, dssPath, t, units, dataType);
     }
 
-    private static void WriteToDss(DssWriter dss, float[] data, string dssPath,DateTime startTime, string units, string dataType)
+    private static void WriteToDss(DssWriter dss, float[] data, string dssPath, DateTime startTime, string units, string dataType)
     {
-      double[] d = new double[data.Length];
-      Array.Copy(data, d, d.Length);
-      // insert leap days... copy prev value...
+      double[] d = addLeapDays(data, startTime);
       Hec.Dss.TimeSeries ts = new TimeSeries(dssPath, d, startTime, units, dataType);
-      
-      try{
+
+      try
+      {
         dss.Write(ts, true);
-      }catch(AccessViolationException e){
+      }
+      catch (AccessViolationException e)
+      {
         Console.WriteLine("Access Violation Occurred, trying again. " + e.Message);
-        try{
+        try
+        {
           dss.Write(ts, true);
-        }catch(AccessViolationException e2){
-          File.WriteAllText("WriteExceptions.txt","Exception " + e2.Message + " " + dssPath);
         }
-      }catch(Exception ex){
+        catch (AccessViolationException e2)
+        {
+          File.WriteAllText("WriteExceptions.txt", "Exception " + e2.Message + " " + dssPath);
+        }
+      }
+      catch (Exception ex)
+      {
         Console.WriteLine("Exception " + ex.Message + " " + dssPath);
       }
     }
 
+    private static double[] addLeapDays(float[] inputdata, DateTime inputT)
+    {
+      int totalCount = inputdata.Length;
+      List<double> output = new List<double>(totalCount + 24 * inputdata.Length / 24 / 364);
+      int nonLeapDayIndex = 0;
+      int leapDayIndex = 0;
+      float defaultValue = 0.0f;
+      DateTime t = inputT;
+      while (nonLeapDayIndex < inputdata.Length)
+      {
+        if (t.Month == 12 && t.Day == 31 && DateTime.IsLeapYear(t.Year))
```

Comment:
since we know we are adding exactly 365*24 values for every year why can't we just add all of those without any condition checking?

Only in the condition that we know it is LeapYear we just add 24 default values before proceeding. It seems like there would be alot less condition checking and data manipulation

Reply:
I was shooting for super simple to read code.   If we need more performance we can setup some metrics.  It is likely disk IO dominates in the app?

Reply:
understood. lets go with your soultion